android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -252,41 +237,21 @@ public void onBackPressed() {
     finish();
   }
 
-  private static int getCurrentKeyboardIndex() {
+  /**
+   * Get the current keyboard index. Only to use for KeyboardPickerActivity or unit test.
+   * @return int of the matching keyboard index. If current index not found, return 0
+   */
+  public static int getCurrentKeyboardIndex() {
     int pos = 0;
+    pos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
```

Comment:
```suggestion
    int pos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
```

Reply:
> Are there cases where it might return 0 but there are actually no keyboards in the list?

The two scenarios I can think of:
1. If an app never calls `KMManager.addKeyboard()` (in MainActivity / SystemKeyboard
2. In the FV app, there's a window of time when the app rebuilds the entire keyboards list (removes all existing ones, then adds all the selected keyboards)

Reply:
So in those cases, this will return an invalid index of 0 when it should probably return -1. If it's gonna crash, this function should at least not be lying about what indexes are valid :)

Reply:
ok - given that I've removed KeyboardPickerActivity.getKeyboardIndex() since it's all handled in KeyboardController